python/lsst/dax/imgserv/vo/imageSODA.py

```diff
@@ -50,47 +58,64 @@ class ImageSODA(SODA):
     def __init__(self, config):
         self._config = config
 
-    def do_sync(self, params: dict) -> object:
+    def do_sync(self, params: dict) -> afw_image:
         """ Do sync operation.
 
         Parameters
         ---------
         params : `dict`
-            the HTTP parameters
+            the request parameters (See Shape requirements above)
+
+        Returns
+        -------
+        resp: `lsst.afw.image`
+            the image object.
         """
         if "POS" in params:
             resp = super().handle_pos(params)
             if isinstance(resp, tuple):
-                # image is first element of the tuple
+                # image object is first element of the tuple
                 return resp[0]
             else:
                 return resp
         else:
             raise NotImplementedError("ImageSODA.do_sync(): Unsupported "
                                       "Request")
 
-    def do_async(self, params: dict) -> object:
-        """ Do async operation.
+    def do_async(self, params: dict) -> str:
+        """ For async operation, create a new task for the request, enqueue for
+        later processing, then return the task_id for tracking it.
 
         Parameters
         ----------
         params : `dict`
-            the HTTP parameters.
+            the request parameters. (See Shape parameters above)
 
         Returns
         -------
-        xml: `str`
+        task.task_id: `str`
+            the newly created task/job id.
 
         """
-        raise NotImplementedError("ImageSODA.do_async()")
+        user = session.get("user", "UNKNOWN")
+        # enqueue the request for image_worker
+        job_start_time = datetime.timestamp(datetime.now())
```

Comment:
You might want to set the start time in the worker part, not when the request is received.  Otherwise your execution time (end time - start time) is how long it took to process and get through the queue, which isn't nearly as useful as just the runtime.

The time you're doing here should be the creation time, which I don't see in your UWS template, but it's in the spec:
<uws:creationTime>2009-05-19T17:01:44.038Z</uws:creationTime>
http://www.ivoa.net/documents/UWS/20161024/REC-UWS-1.1-20161024.pdf page 14

Reply:
Very good point - implemented as described.